### PR TITLE
list:repeat

### PR DESCRIPTION
**Pull request by bsnyder788**

### CODE REVIEW

File: gleam_stdlib/src/list.gleam [hunk 1]
```diff
@@ -263,3 +263,12 @@ pub fn range(start, stop) {
   | order:Lt -> [start | range(start + 1, stop)]
   }
 }
+
+external fn duplicate(t, a) -> List(a) = "lists" "duplicate"
+
+pub fn repeat(a, times) {
+  case times <= 0 {
+    | True -> []
+    | False -> duplicate(times, a)
```
Comment (lpil) [inline gleam_stdlib/src/list.gleam]: Can we implement this in Gleam rather than calling out to Erlang? :)

Comment (bsnyder788) [reply gleam_stdlib/src/list.gleam]: sure! wasn't sure what the policy was on using Erlang for some of these

File: gleam_stdlib/test/list_test.gleam [hunk 2]
```diff
@@ -298,3 +298,17 @@ pub fn range_test() {
...
+  |> expect:equal(_, [])
+
+  list:repeat(1, 0)
+  |> expect:equal(_, [])
+
+  list:repeat(2, 3)
+  |> expect:equal(_, [2, 2, 2])
+
+  list:repeat("x", 5)
+  |> expect:equal(_, ["x","x","x","x","x"])
```
Comment (lpil) [inline gleam_stdlib/test/list_test.gleam]: Could we have spaces after the `,`s please? Thank you

Comment (bsnyder788) [reply gleam_stdlib/test/list_test.gleam]: fixed!

File: gleam_stdlib/src/list.gleam (hunk 1 above)
Comment (bsnyder788) [reply gleam_stdlib/src/list.gleam]: reimplemented in Gleam :)

**Comment by lpil** [comment]
I generally think we should implement things in Gleam unless in Erlang it is implemented using a BIF or syntax that Gleam does not support. :)

**Pull request merged**